src/Scratch.jl

```diff
@@ -180,6 +180,12 @@ collected.  See `Pkg.gc()` and `track_scratch_access()` for more details.
 """
 function get_scratch!(key::AbstractString, parent_pkg::Union{Module,UUID,Nothing} = nothing,
                                            calling_pkg::Union{Module,UUID,Nothing} = parent_pkg)
+    # Verify that the key is valid (only needed here at construction time)
+    if match(r"^[a-zA-Z0-9-\.]+$", key) === nothing
```

Comment:
Not sure what else might be useful. I was thinking `.` is nice if you want to have a version number folder (e.g. `1.0.0`) and that `-` is common instead of space for multiword keys.